src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -1735,5 +1759,63 @@ private RendezvousAwaitable<int> InternalReceiveResponseHeadersAsync(WinHttpRequ
 
             return state.LifecycleAwaitable;
         }
+
+        internal void ChangeCleanerTimer(TimeSpan timeout)
+        {
+            Debug.Assert(_certificateCleanupTimer != null);
+            if (_certificateCleanupTimer!.Change(timeout, Timeout.InfiniteTimeSpan))
+            {
+                isTimerRunning = timeout != Timeout.InfiniteTimeSpan;
+            }
```

Comment:
This is prone to race conditions, e.g. you could currently end up in a state where a certificate is added, but a different thread disabled the timer.
`HttpConnectionPoolManager` uses a lock around these modifications.